src/test/java/lambda/part3/exercise/Exercise2.java

```diff
@@ -34,51 +37,66 @@ private MapHelper(List<T> source) {
          * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
          * Для добавления в новый список каждый элемент преобразовывается с использованием заданной функции.
          * ([T], (T -> R)) -> [R]
+         *
          * @param mapping Функция преобразования элементов.
          */
         public <R> MapHelper<R> map(Function<T, R> mapping) {
-            throw new UnsupportedOperationException();
+            List<R> result = new ArrayList<>();
+            for (T elem : source) {
+                result.add(mapping.apply(elem));
+            }
+            return new MapHelper<>(result);
         }
 
         /**
          * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
          * Для добавления в новый список каждый элемент преобразовывается в список с использованием заданной функции.
          * ([T], (T -> [R])) -> [R]
+         *
          * @param flatMapping Функция преобразования элементов.
          */
         public <R> MapHelper<R> flatMap(Function<T, List<R>> flatMapping) {
-            throw new UnsupportedOperationException();
+            List<List<R>> lists = new ArrayList<>();
+            for (T elem : source) {
+                lists.add(flatMapping.apply(elem));
+            }
+
+            List<R> result = new ArrayList<>();
+            for (List<R> list : lists) {
+                result.addAll(list);
+            }
+
+            return new MapHelper<>(result);
         }
```

Comment:
Lines 59-59 could be shortened to 3 lines with forEach, andThen and method references.